File: fc-chat/demo/part3/ch2/03-security-login-logout-websocket/src/main/java/net/prostars/messagesystem/config/WebSocketConfig.java
package net.prostars.messagesystem.config;

import net.prostars.messagesystem.auth.CustomHttpSessionHandshakeInterceptor;
import net.prostars.messagesystem.handler.WebSocketHandler;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.socket.config.annotation.EnableWebSocket;
import org.springframework.web.socket.config.annotation.WebSocketConfigurer;
import org.springframework.web.socket.config.annotation.WebSocketHandlerRegistry;

@Configuration
@EnableWebSocket
public class WebSocketConfig implements WebSocketConfigurer {

  private final WebSocketHandler webSocketHandler;
  private final CustomHttpSessionHandshakeInterceptor customHttpSessionHandshakeInterceptor;

  public WebSocketConfig(
      WebSocketHandler webSocketHandler,
      CustomHttpSessionHandshakeInterceptor customHttpSessionHandshakeInterceptor) {
    this.webSocketHandler = webSocketHandler;
    this.customHttpSessionHandshakeInterceptor = customHttpSessionHandshakeInterceptor;
  }

  @Override
  public void registerWebSocketHandlers(WebSocketHandlerRegistry registry) {
    registry
        .addHandler(webSocketHandler, "/ws/v1/connect");
//        .addInterceptors(customHttpSessionHandshakeInterceptor);
  }
}

File: fc-chat/demo/part2/preview-websocket/src/main/kotlin/com/yscorp/previewwebsocket/timer/config/WebSocketConfig.kt
package com.yscorp.previewwebsocket.timer.config

import com.yscorp.previewwebsocket.timer.handler.TimerHandler
import org.springframework.web.socket.config.annotation.EnableWebSocket
import org.springframework.web.socket.config.annotation.WebSocketConfigurer
import org.springframework.web.socket.config.annotation.WebSocketHandlerRegistry

@org.springframework.context.annotation.Configuration
@EnableWebSocket
class WebSocketConfig : WebSocketConfigurer {
    override fun registerWebSocketHandlers(registry: WebSocketHandlerRegistry) {
        registry.addHandler(TimerHandler(), "/ws/timer")
    }
}
